ngraph/test/threading.cpp

```diff
@@ -60,3 +62,26 @@ TEST(threading, get_friendly_name) {
         th.join();
     }
 }
+
+TEST(threading, check_atomic_guard) {
+    std::atomic_bool test_val{false};
+    int result = 2;
+    const auto& thread1_fun = [&]() {
+        ov::AtomicGuard lock(test_val);
+        std::chrono::milliseconds ms{2000};
+        std::this_thread::sleep_for(ms);
+        result += 3;
+    };
+    const auto& thread2_fun = [&]() {
+        ov::AtomicGuard lock(test_val);
```

Comment:
This test assumes that `thread1_fun` will start execution before `thread2_fun` which is not guaranteed

Reply:
Added a timeout before locking for thread2.